pkg/check/datastore.go

```diff
@@ -226,16 +234,111 @@ func getPolicy(ctx *CheckContext, name string) ([]types.BasePbmProfile, error) {
 	return c.RetrieveContent(tctx, []types.PbmProfileId{{UniqueId: name}})
 }
 
-func checkDataStore(dsName string, infrastructure *configv1.Infrastructure) error {
+func checkDataStore(ctx *CheckContext, dsName string, infrastructure *configv1.Infrastructure) error {
 	clusterID := infrastructure.Status.InfrastructureName
 	volumeName := fmt.Sprintf("[%s] 00000000-0000-0000-0000-000000000000/%s-dynamic-pvc-00000000-0000-0000-0000-000000000000.vmdk", dsName, clusterID)
 	klog.V(4).Infof("Checking data store %q with potential volume Name %s", dsName, volumeName)
 	if err := checkVolumeName(volumeName); err != nil {
 		return fmt.Errorf("datastore %s: %s", dsName, err)
 	}
+	if err := checkForDatastoreCluster(ctx, dsName); err != nil {
+		return err
+	}
 	return nil
 }
 
+func checkForDatastoreCluster(ctx *CheckContext, dataStoreName string) error {
```

Comment:
Perhaps for a subsequent PR: This is called for every StorageClass and loads non-trivial amount of data from vSphere. Would it be possible to call it only once per `dataStorageName`? I.e. with a cache _somewhere_ in `CheckStorageClasses`? It's not that easy, because there is CheckDefaultDatastore too...

Reply:
Yes we could. I think it might be worth caching the information within the same run. but once all the checks are run -we can automatically expire the cache, so as they aren't reused in next run.